src/contexts/login-context.js

```diff
@@ -0,0 +1,93 @@
+import axios from "axios";
+import { createContext, useContext, useReducer, useState } from "react";
+import { useNavigate } from "react-router-dom";
+
+const LoginContext = createContext("");
+const defaultCredentialTemplate = {
+    email: "",
+    password: ""
+}
+
+const userInfoTemplate = {
+    details: "",
+    encodedToken: ""
+};
+
+const loginReducer = (userCredentials, action) => {
+    switch(action.type) {
+        case "EMAIL" :
+            return {...userCredentials, email : action.payload}
+        case "PASSWORD" : 
+            return {...userCredentials, password : action.payload}
+        default: return ({...defaultCredentialTemplate})
+    }
+}
+
+const userInfoReducer = (userInfo, action) => {
+    switch(action.type) {
+        case "DETAILS" : 
+            return {...userInfo, details: action.payload}
+        case "ENCODED_TOKEN" :
+            return {...userInfo, encodedToken: action.payload}
+        default: return ({...userInfoTemplate})
+    }
+}
+
```

Comment:
Do shift this entire block of code to a separate file called the authReducers having these making it easy to understand the work you have done in the context.

Reply:
thankyou so much @dev-enforced for your suggestions. I have made some changes accordingly